src/virtual_renderer.js

```diff
@@ -1902,6 +1922,40 @@ class VirtualRenderer {
         this.updateLines(row, row);
     }
 
+    // Hide all non-ghost-text tokens to the right of a given position.
+    hideTokensAfterPosition(row, column) {
+        var tokens = this.session.getTokens(row);
+        var l = 0;
+        var hasPassedCursor = false;
+        var hiddenTokens = [];
+        // Loop over all tokens and track at what position in the line they end.
+        for (var i = 0; i < tokens.length; i++) {
+            var token = tokens[i];
+            l += token.value.length;
+
+            if (token.type === "ghost_text") continue;
+
+            // If we've already passed the current cursor position, mark all of them as hidden.
+            if (hasPassedCursor) {
+                hiddenTokens.push({type: token.type, value: token.value});
+                token.type = "hidden_token";
+                continue;
+            }
+            if (l === column) {
+                hasPassedCursor = true;
+            } else if (l > column) {
+                hasPassedCursor = true;
+                var diff = l - column;
+                var before = token.value.slice(0, token.value.length - diff);
+                var after = token.value.slice(token.value.length - diff);
+                tokens.splice(i, 1, {type: token.type, value: before}, {type: "hidden_token", value: after});
+                hiddenTokens.push({type: token.type, value: after});
+            }
```

Comment:
Ah yeah good point. Before I did some refactoring, this method was called before the new token (the ghost text on the cursor line) was inserted. Then it could be the case that the token the cursor was in had to be split. But now that we call the method after inserting the ghost text token, that splitting is already done by the `addToken` method so this isn't used anymore.

Reply:
Yeah, that's in line with what I'm seeing
![Screenshot 2024-07-22 at 14 25 18](https://github.com/user-attachments/assets/36398332-d22a-41fb-803d-d3522eadb03c)
: